### PR TITLE
Ldanzinger/densify generalize

### PR DESCRIPTION
**Pull request by ldanzinger**
@michael-tims please review/merge

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Geometry/DensifyAndGeneralize/DensifyAndGeneralize.cpp [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  MultipointBuilder multipointBuilder(pointCollection->spatialReference());
+  multipointBuilder.setPoints(pointCollection);
+  SimpleMarkerSymbol* originalSms = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Circle, QColor("red"), 7.0f /*size*/, this);
+  m_originalMultipointGraphic = new Graphic(multipointBuilder.toGeometry(), originalSms, this);
+  m_originalMultipointGraphic->setZIndex(0);
+  m_graphicsOverlay->graphics()->append(m_originalMultipointGraphic);
+
+  // original red dotted line graphic
+  PolylineBuilder polylineBuilder(pointCollection->spatialReference());
+  for (int i = 0; i < pointCollection->size(); i++)
```
Comment (michael-tims) [inline ArcGISRuntimeSDKQt_CppSamples/Geometry/DensifyAndGeneralize/DensifyAndGeneralize.cpp]:
```
const int pointCollectionSize = pointCollection->size();
for (int i = 0; i < pointCollectionSize; i++)
```

File: ArcGISRuntimeSDKQt_CppSamples/Geometry/DensifyAndGeneralize/DensifyAndGeneralize.cpp [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    polyline = GeometryEngine::densify(polyline, maxSegmentLength);
+
+  // Update the line graphic
+  m_resultLineGraphic->setGeometry(polyline);
+
+  // Update the multipoint graphic
+  MultipointBuilder multipointBuilder(polyline.spatialReference());
+  PointCollection* pointCollection = new PointCollection(polyline.spatialReference(), this);
+  if (polyline.parts().size() < 1)
+    return;
```
Comment (michael-tims) [inline ArcGISRuntimeSDKQt_CppSamples/Geometry/DensifyAndGeneralize/DensifyAndGeneralize.cpp]: Add new line after `return;`

File: ArcGISRuntimeSDKQt_CppSamples/Geometry/DensifyAndGeneralize/DensifyAndGeneralize.cpp [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  ImmutablePointCollection polylinePoints = polyline.parts().part(0).points();
+  for (int i = 0; i < polylinePoints.size(); i++) {
```
Comment (michael-tims) [inline ArcGISRuntimeSDKQt_CppSamples/Geometry/DensifyAndGeneralize/DensifyAndGeneralize.cpp]:
```
const int polylinePointsSize = polylinePoints.size();
for (int i = 0; i < polylinePointsSize ; i++)
```

Comment (michael-tims) [reply ArcGISRuntimeSDKQt_CppSamples/Geometry/DensifyAndGeneralize/DensifyAndGeneralize.cpp]: Brace on new line

File: ArcGISRuntimeSDKQt_CppSamples/Geometry/DensifyAndGeneralize/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+![](screenshot.png)
+
+## How to use the sample
+Use the sliders to control the parameters of the densify and generalize methods. You can deselect the checkboxes for either method to remove its effect from the result polyline. You can also hide the result to only see the original by unchecking the "Show result" checkbox.
+
+## How it works
+To densify and generalize a polyline:
+- Use the static method GeometryEngine::densify(polyline, maxSegmentLength) to densify the polyline. The resulting polyline will add points along the line so that there are no points greater than maxSegmentLength from the next point.
+- Use the static method GeometryEngine::generalize(polyline, maxDeviation, true) to generalize the polyline. The resulting polyline will have points or shifted from the line to simplify the shape. None of these points can deviate farther from the original line than maxDeviation. The last parameter, removeDegenerateParts , will clean up extraneous parts if the geometry is multi-part (it will have no effect in this sample.
```
Comment (michael-tims) [inline ArcGISRuntimeSDKQt_CppSamples/Geometry/DensifyAndGeneralize/README.md]: `resulting polyline will have points or shifted from the line...`  seems like a word got added or is missing

Comment (michael-tims) [reply ArcGISRuntimeSDKQt_CppSamples/Geometry/DensifyAndGeneralize/README.md]: `removeDegenerateParts ,` -> `removeDegenerateParts,`

Comment (michael-tims) [reply ArcGISRuntimeSDKQt_CppSamples/Geometry/DensifyAndGeneralize/README.md]: `in this sample.` -> `in this sample).`

File: ArcGISRuntimeSDKQt_CppSamples/Geometry/DensifyAndGeneralize/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+- Note that maxSegmentLength and maxDeviation are in the units of geometry's coordinate system. This could be in degrees in some coordinate systems. In this example, a cartesian coordinate system is used and at a small enough scale that geodesic distances are not required.
```
Comment (michael-tims) [inline ArcGISRuntimeSDKQt_CppSamples/Geometry/DensifyAndGeneralize/README.md]: `that geodesic distances are not required.` -> `the geodesic distances are not required.`?

File: ArcGISRuntimeSDKQt_QMLSamples/Geometry/DensifyAndGeneralize/DensifyAndGeneralize.qml [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+    width: 800
+    height: 600
+
+    property real scaleFactor: System.displayScaleFactor
+
+    MapView {
+        id: mapView
+        anchors.fill: parent
+
+
```
Comment (michael-tims) [inline ArcGISRuntimeSDKQt_QMLSamples/Geometry/DensifyAndGeneralize/DensifyAndGeneralize.qml]: Remove extra blank line

**Comment by ldanzinger** [comment]
@michael-tims thanks for the review. this should be updated

**Pull request merged**